packages/xo-server/src/api/host.js

```diff
@@ -176,7 +176,12 @@ forget.resolve = {
 // Returns an empty array if up-to-date
 // Throws an error if the host is not running the latest XS version
 export function listMissingPatches ({ host }) {
-  return this.getXapi(host).listMissingPoolPatchesOnHost(host._xapiId)
+  // XCP-ng handles update differently than XenServer
+  if (host.productBrand === 'XCP-ng') {
+    return this.getXapi(host).xcpListHostUpdates(host._xapiId)
+  } else {
+    return this.getXapi(host).listMissingPoolPatchesOnHost(host._xapiId)
+  }
```

Comment:
Don't you think it would be best to move the test directly into `listMissingPoolPatchesOnHost`?

Reply:
The patching process is vastly different (than XenServer). I'm afraid that going deeper in the backend the result will be more complex in the end. But it's just my intuition, if you want to re-design the stuff, it's up to you :+1: